grc/core/utils/extract_docs.py

```diff
@@ -246,27 +248,31 @@ def terminate(self):
 def worker_main():
     """
     Main entry point for the docstring extraction process.
-    Manages RPC with main process through.
+    Manages RPC with main process through stdin/stdout.
     Runs a docstring extraction for each key it read on stdin.
     """
-    def send(cmd, args):
+    def send(code, cmd, args):
         json.dump((code, cmd, args), sys.stdout)
-        sys.stdout.write('\n'.encode())
+        sys.stdout.write('\n')
+        # fluh out to get new commands from the queue into stdin
+        sys.stdout.flush()
 
     sys.stdout.write('1')
+    # flush out to signal the main process we are ready for new commands
+    sys.stdout.flush()
     for line in iter(sys.stdin.readline, ''):
         code, cmd, args = json.loads(line, encoding='utf-8')
         try:
             if cmd == 'query':
                 key, imports, make = args
-                send('result', (key, docstring_from_make(key, imports, make)))
+                send(code, 'result', (key, docstring_from_make(key, imports, make)))
```

Comment:
Can remove the code part again, but it wasn't super obvious that it's enclosed already and debugging the worker wasn't really easy. If we can drop py2.6 for 3.8/next we should switch to a `multiprocessing` based approach with IPC pipes instead of string parsing of stdin/out.

Reply:
Sure, but some de-/serialization will happen anyway. Usually there are more pressing issues though

Reply:
Debugging the Py3k bytes/text issue was kinda annoying though since the IPC is handled through stdin/out.

Reply:
we can drop Py2.6 from master right now if you need that, it's defintely not something we need to support in 3.8

Reply:
NB: "Drop Py2.6" doesn't mean "Drop Py2".